### PR TITLE
Fix binary secret update

### LINKED ISSUE #808: Implement Raw mode for updating secrets
## Expected Behaviour

Once created, a raw secret file (created via faas-cli create), can be updated via the update command / API and retain its binary format.

## Current Behaviour

Create / delete / create is currently required for binary secret data.

## List All Possible Solutions and Workarounds

Change the "update" handler code to match the "create" code, so that raw data can be retained, without reverting to a string format.

## Steps to Reproduce (for bugs)

1. `faas-cli secret create NAME --from-file file.zip`
2. `faas-cli secret update NAME --from-file file.zip`

Now, `file.zip` has been stored as a string instead of Raw binary data.

## Context

This is part of the feature merged in #806, but was missed.

### PR DESCRIPTION
**Pull request by nitishkumar71**
Signed-off-by: Nitishkumar Singh <nitishkumarsingh71@gmail.com>

fix update secret test

Signed-off-by: Nitishkumar Singh <nitishkumarsingh71@gmail.com>

## Description

Closes #808 
## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

## How Has This Been Tested?

1. Create Zip file using `zip test.zip file1.txt`
2. Create a Secret
```
faas-cli secret create test --from-file test.zip 

WARNING! You are not using an encrypted connection to the gateway, consider using HTTPS.
Creating secret: test
Created: 202 Accepted
```
3. Describe Secret
```
kubectl describe secret/test -n openfaas-fn
Name:         test
Namespace:    openfaas-fn
Labels:       app.kubernetes.io/managed-by=openfaas
Annotations:  <none>

Type:  Opaque

Data
====
test:  265 bytes
```
4. Print Secret. The output is being printed in binary format
```
kubectl get secret/test -n openfaas-fn -o jsonpath="{.data.test}" | base64 -d;echo
```
5. Update zip file using `zip -u test.zip file2.txt`
6. Update Secret using faas-cli
```
faas-cli secret update test --from-file test.zip 
WARNING! You are not using an encrypted connection to the gateway, consider using HTTPS.
Updating secret: test
Updated: 202 Accepted
```
7. Describe Secret
```
kubectl describe secret/test -n openfaas-fn
Name:         test
Namespace:    openfaas-fn
Labels:       app.kubernetes.io/managed-by=openfaas
Annotations:  <none>

Type:  Opaque

Data
====
test:  9217 bytes
```
8. Print Secret. The output is being printed in binary format
```
kubectl get secret/test -n openfaas-fn -o jsonpath="{.data.test}" | base64 -d;echo
```
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by alexellis — DISMISSED** [review]
Looks good to me. Thanks for picking this up, it was something I had on my plate but glad that someone got there first.

**Comment by alexellis** [comment]
CI is failing on this PR, but also on others.

@nitishkumar71, if you have a moment would you mind looking into it? I don't think it's related to your changes. KinD is timing out starting-up.

**Comment by nitishkumar71** [comment]
> CI is failing on this PR, but also on others.
> 
> @nitishkumar71, if you have a moment would you mind looking into it? I don't think it's related to your changes. KinD is timing out starting-up.

@alexellis  Sorry wasn't able to come online for few days.

I debugged into it and looks like CNI plugin is not being initialized for the kind cluster. The issue only occurs in Github actions. I can create cluster using script on my local.

https://github.com/nitishkumar71/faas-netes/runs/3012661037?check_suite_focus=true

File: pkg/k8s/secrets.go [hunk 1]
```diff
@@ -187,6 +178,20 @@ func (c secretClient) validateSecret(secret types.Secret) error {
...
+		data = map[string][]byte{
+			secret.Name: secret.RawValue,
+		}
+	} else {
+		data = map[string][]byte{
+			secret.Name: []byte(secret.Value),
+		}
+	}
+	return data
+}
```
Comment (LucasRoesler) [inline pkg/k8s/secrets.go:193]: Early returns, like below, are generally preferred to if/else, you can even remove the placeholder `data` variable in this case 
```suggestion
	if len(secret.RawValue) > 0 {
		return map[string][]byte{
			secret.Name: secret.RawValue,
		}
	}
	
	return map[string][]byte{
		secret.Name: []byte(secret.Value),
	}
}
```

Comment (nitishkumar71) [reply pkg/k8s/secrets.go]: Thanks for the feedback.

**Review by LucasRoesler — APPROVED** [review]

**Comment by alexellis** [comment]
Merging. Thank you.

**Pull request merged**